frontend/src/js/components/PageViewer/PageViewer.tsx

```diff
@@ -40,126 +46,52 @@ export const PageViewer: FC<PageViewerProps> = () => {
       .then((obj) => setTotalPages(obj.pageCount));
   }, [uri]);
 
-  // Keypress overrides
-  const handleUserKeyPress = useCallback((e) => {
-    if ((e.ctrlKey || e.metaKey) && e.keyCode === 70) {
-      e.preventDefault();
-      setFindVisible(true);
-
-      const maybeInput = document.getElementById(
-        "find-search-input"
-      ) as HTMLInputElement;
-      if (maybeInput) {
-        maybeInput.focus();
-        maybeInput.setSelectionRange(0, maybeInput.value.length);
-      }
-    }
-  }, []);
-
-  // Register keypress overrides
   useEffect(() => {
-    window.addEventListener("keydown", handleUserKeyPress);
-    return () => {
-      window.removeEventListener("keydown", handleUserKeyPress);
-    };
-  }, [handleUserKeyPress]);
-
-  const performFind = useCallback(
-    (query: string) => {
-      const params = new URLSearchParams();
-      // The backend will respect quotes and do an exact search,
-      // but if quotes are unbalanced elasticsearch will error
-      params.set("fq", removeLastUnmatchedQuote(query));
-
-      setIsFindPending(true);
-      return authFetch(`/api/pages2/${uri}/find?${params.toString()}`)
-        .then((res) => res.json())
-        .then((highlights) => {
-          setIsFindPending(false);
-          setFindHighlights(highlights);
-          if (highlights.length) {
-            setFocusedFindHighlightIndex(0);
-          } else {
-            setFocusedFindHighlightIndex(null);
-          }
-          setTriggerRefresh((t) => t + 1);
-        })
-    },
-    [uri]
-  );
-
-  const preloadNextPreviousFindPages = useCallback((
-    focusedFindHighlightIndex: number,
-    findHighlights: HighlightForSearchNavigation[]
-  ) => {
-    const length = findHighlights.length;
-
-    const indexesOfHighlightsToPreload = uniq(
-      range(-3, 3).map((offset) => {
-        const offsetIndex = focusedFindHighlightIndex + offset;
-        // modulo - the regular % is 'remainder' in JS which is different
-        return ((offsetIndex % length) + length) % length;
-      })
-    );
-
-    const newPreloadPages = uniq(indexesOfHighlightsToPreload.map(
-      (idx) => findHighlights[idx].pageNumber
-    ));
-
-    setPageNumbersToPreload(newPreloadPages);
-  }, []);
-
-  const jumpToNextFindHit = useCallback(() => {
-    if (findHighlights.length > 0) {
-      const nextHighlightIndex = (focusedFindHighlightIndex !== null && focusedFindHighlightIndex < (findHighlights.length - 1))
-          ? (focusedFindHighlightIndex + 1)
-          : 0;
-
-      preloadNextPreviousFindPages(nextHighlightIndex, findHighlights);
-      setFocusedFindHighlightIndex(nextHighlightIndex);
+    if (findHighlightsState.focusedIndex !== null && findHighlightsState.highlights.length) {
+      const length = findHighlightsState.highlights.length;
+
+      const indexesOfHighlightsToPreload = uniq(
+          range(-3, 3).map((offset) => {
```

Comment:
maybe comment explaining what -3 and 3 represent, it's pretty obvious though so not totally necessary (I assume the number of highlights in each direction to preload)